middleware/dnstap/msg/msg.go

```diff
@@ -87,3 +95,29 @@ func (d *Data) ToClientQuery() *tap.Message {
 		QueryPort:      &d.Port,
 	}
 }
+
+// Transform the data into a forwader or resolver query message.
```

Comment:
Yes! `msg` is a helper package, `ToOutsideQuery` may be useful for other middlewares.